lib/generate.js

```diff
@@ -45,14 +45,26 @@ var typeFactory = new TypeFactory();
 
 var Generator = function() {
     var generate = function(schema, callback) {
-        typeFactory.gimme(schema, function(err, tf) {
-            if (err) {
-                return callback(err, null);
-            }
-            tf.handle(schema, function(err, val) {
-                callback(err, val);
+        var promise = new Promise(function(resolve, reject) {
```

Comment:
This makes use of the native `Promise` in node which was only introduced in v0.12 but was often [not considered stable](https://stackoverflow.com/a/34524086/2140627) until v4.0.0 (https://nodejs.org/en/blog/release/v4.0.0/).